### PR TITLE
feat: scrub Set-Cookie header in debug logs

### LINKED ISSUE #1189: ORAS debug log should scrub Set-Cookie header in the logs
### What is the version of your ORAS CLI

1.1.0

### What would you like to be added?

The [Set-Cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie) header might contains sensitive information and need to be scrubbed from the debug logs.

### Why is this needed for ORAS?

So user won't need to redact the header when pasting the ORAS logs online.

### Are you willing to submit PRs to contribute to this feature?

- [X] Yes, I am willing to implement it.

### PR DESCRIPTION
**Pull request by qweeah**
**What this PR does / why we need it**:
Fixes #1189.

### CODE REVIEW

File: internal/trace/transport.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func logHeader(header http.Header) string {
 	if len(header) > 0 {
 		headers := []string{}
 		for k, v := range header {
-			if strings.EqualFold(k, "Authorization") {
+			if strings.EqualFold(k, "Authorization") || strings.EqualFold(k, "Set-Cookie") {
```
Comment (shizhMSFT) [inline internal/trace/transport.go:71]: Can we define an array for this?

Comment (qweeah) [reply internal/trace/transport.go:71]: Refactored and defined a map of it.

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**